### DISCUSSION TITLE
Are model objects thread safe?

### DISCUSSION THREAD

**Discussion by chitralverma** [qdrant/fastembed#191] [Q&A]
I was planning to use fastembed as part of an API server. The model is pre-cached on the app level and when a POST request arrives, I use the payload as an input to the model.

I observed that with single requests this works okay, but for multiple requests in parallel, there are some non-deterministic latencies.

Any thoughts/ suggestions on this?

**Comment by rahulbknowbe4** [comment]
Were you able to find an answer to this?
Is this behavior documented someplace?
